frontend/OBSStudioAPI.cpp

```diff
@@ -73,14 +73,9 @@ void OBSStudioAPI::obs_frontend_set_current_scene(obs_source_t *scene)
 
 void OBSStudioAPI::obs_frontend_get_transitions(struct obs_frontend_source_list *sources)
 {
-	for (int i = 0; i < main->ui->transitions->count(); i++) {
-		OBSSource tr = main->ui->transitions->itemData(i).value<OBSSource>();
-
-		if (!tr)
-			continue;
-
-		if (obs_source_get_ref(tr) != nullptr)
-			da_push_back(sources->sources, &tr);
+	for (auto &pair : main->transitions) {
+		if (obs_source_get_ref(pair.second) != nullptr)
+			da_push_back(sources->sources, &pair.second);
```

Comment:
```suggestion
	for (auto& [uuid, transition] : transitions) {
		if (obs_source_get_ref(transition) != nullptr) {
			da_push_back(sources->sources, &transition);
		}
	}
```

I'm not sure about passing in a reference to the transition though - that way you add the `OBSSource` object into the dynamic array and not the underlying source object (with the reference count increased).